core/slim_sim.cpp

```diff
@@ -6398,115 +6446,142 @@ void SLiMSim::ReadProvenanceTable(tsk_table_collection_t *p_tables, slim_generat
-		
-		/*
-		try {
-			model_type_str = j["model_type"];
-			gen_ll = j["generation"];
-			//rem_count_ll = j["remembered_node_count"];	// no longer using this key
-		}
-		catch (...)
-		{
-			EIDOS_TERMINATION << "ERROR (SLiMSim::ReadProvenanceTable): error reading provenance value (file_version 0.1); this file cannot be read." << EidosTerminate();
-		}
-		 */
-	}
-	else if (file_version_02.is_string())
-	{
-		// File version 0.2 was before the addition of nucleotides, so MutationMetadataRec_PRENUC will need to be used
-		// File version 0.3 is the current format, with nucleotides, using MutationMetadataRec
-		if (file_version_02 == SLIM_TREES_FILE_VERSION_PRENUC)
-			*p_file_version = 2;
-		else if (file_version_02 == SLIM_TREES_FILE_VERSION_POSTNUC)
-			*p_file_version = 3;
-		else if (file_version_02 == SLIM_TREES_FILE_VERSION)
-			*p_file_version = 4;
-		else
-			EIDOS_TERMINATION << "ERROR (SLiMSim::ReadProvenanceTable): this .trees file was generated by an unrecognized version of SLiM or pyslim; this file cannot be read." << EidosTerminate();
-		
-		try {
-			model_type_str = j["parameters"]["model_type"];
-			gen_ll = j["slim"]["generation"];
-			//rem_count_ll = j["slim"]["remembered_node_count"];	// no longer using this key
-		}
-		catch (...)
-		{
-			EIDOS_TERMINATION << "ERROR (SLiMSim::ReadProvenanceTable): error reading provenance value (file_version " << file_version_02 << "); this file cannot be read." << EidosTerminate();
-		}
-	}
-	else
-	{
-		EIDOS_TERMINATION << "ERROR (SLiMSim::ReadProvenanceTable): missing or corrupted file version; this file cannot be read." << EidosTerminate();
-	}
+
+    ////////////
+    // Format 0.5: using top-level metadata
+    try {
+        // Note: we *could* parse the metadata schema (which is json),
+        // but instead we'll just try parsing the metadata.
+        // std::string metadata_schema_str(p_tables->metadata_schema, p_tables->metadata_schema_length);
+        // nlohmann::json metadata_schema = nlohmann::json::parse(metadata_schema_str);
+
+        std::string metadata_str(p_tables->metadata, p_tables->metadata_length);
+        auto metadata = nlohmann::json::parse(metadata_str);
+        model_type_str = metadata["SLiM"]["model_type"];
+        gen_ll = metadata["SLiM"]["generation"];
+        auto file_version_03 = metadata["SLiM"]["file_version"];
+		if (file_version_03 == SLIM_TREES_FILE_VERSION_NEW)
+			*p_file_version = 5;
+        else
+			EIDOS_TERMINATION << "ERROR (SLiMSim::ReadTreeSequenceMetadata): this .trees file was generated by an unrecognized version of SLiM or pyslim; this file cannot be read." << EidosTerminate();
+    } catch (...) {
+    ///////////////////////
+    // Previous formats: everything is in provenance
+        // NOTE: maybe we don't want to even throw a warning here?
```

Comment:
I would say no warning, unless there is some problem with reading older files

Reply:
I agree - but I wanted to leave it in for now to make sure that we're actually using the metadata, not just the provenance entry. (Hm, I'll test that by clearing the provenance table in pyslim, though.)

Reply:
I put a test in pyslim that this SLiM can still reload even if the provenance table is cleared. It can!

Reply:
So now that this has been tested, the warning should be removed, yes?

Reply:
I fixed this already.

Reply:
OK.  I haven't found a way to see these comment threads associated with the current diffs instead of the outdated diffs, so it's quite difficult to tell what has been fixed, and how it has been fixed.  I can see the current diffs instead – but then, without the associated conversation.  Grr.